### PR TITLE
SIP-86 Chainlink aggregatorV2V3

### PR DESCRIPTION
**Pull request by clementbalestrat**

### CODE REVIEW

File: contracts/ExchangeRates.sol [hunk 1]
```diff
@@ -12,7 +12,7 @@ import "./SafeDecimalMath.sol";
 
 // Internal references
 // AggregatorInterface from Chainlink represents a decentralized pricing network for a single currency key
-import "@chainlink/contracts-0.0.9/src/v0.5/interfaces/AggregatorInterface.sol";
+import "./interfaces/AggregatorV2V3Interface.sol";
```
Comment (clementbalestrat) [inline contracts/ExchangeRates.sol:15]: @justinjmoses just waiting for chainlink to add it to their repo

Comment (justinjmoses) [reply contracts/ExchangeRates.sol:15]: It's done. Please upgrade to `0.0.10`

File: contracts/ExchangeRates.sol [hunk 2]
```diff
@@ -160,21 +160,23 @@ contract ExchangeRates is Owned, SelfDestructible, MixinResolver, MixinSystemSet
     }
 
     function addAggregator(bytes32 currencyKey, address aggregatorAddress) external onlyOwner {
-        AggregatorInterface aggregator = AggregatorInterface(aggregatorAddress);
+        AggregatorV2V3Interface aggregator = AggregatorV2V3Interface(aggregatorAddress);
         // This check tries to make sure that a valid aggregator is being added.
         // It checks if the aggregator is an existing smart contract that has implemented `latestTimestamp` function.
-        require(aggregator.latestTimestamp() >= 0, "Given Aggregator is invalid");
+        aggregator.latestRoundData();
```
Comment (justinjmoses) [inline contracts/ExchangeRates.sol:166]: you don't want to put a require in here to be more explicit?

Comment (justinjmoses) [reply contracts/ExchangeRates.sol:166]: Also, should we not `require` that `decimals` is <= 18?

File: test/contracts/ExchangeRates.js [hunk 3]
```diff
@@ -22,7 +22,7 @@ const {
 
 const { toBN } = require('web3-utils');
 
-const MockAggregator = artifacts.require('MockAggregator');
+const MockAggregator = artifacts.require('MockAggregatorV2V3');
```
Comment (justinjmoses) [inline test/contracts/ExchangeRates.js:25]: Is there any point keeping the old `MockAggregator` around?

Comment (justinjmoses) [reply test/contracts/ExchangeRates.js:25]: It's used by the publish tests, which are currently breaking: https://app.circleci.com/pipelines/github/Synthetixio/synthetix/3441/workflows/31d93e6b-bdf5-41a9-a2e6-c5cfcd110a65/jobs/19683

**Review by justinjmoses — CHANGES_REQUESTED** [review]
Comments inline

**Comment by justinjmoses** [comment]
Also @clementbalestrat can you make sure you add all updated contracts to `publish/releases.json` so they are marked for release?

File: contracts/test-helpers/MockAggregatorV2V3.sol [hunk 4]
```diff
@@ -0,0 +1,116 @@
+pragma solidity ^0.5.16;
+
+
+interface AggregatorV2V3Interface {
```
Comment (justinjmoses) [inline contracts/test-helpers/MockAggregatorV2V3.sol:4]: So you can simply use `import "@chainlink/contracts-0.0.10/src/v0.5/interfaces/AggregatorV2V3Interface.sol";
` instead, however if you do, then you'll need to stub out everything in the mock. Which would you prefer?

File: contracts/test-helpers/MockAggregatorV2V3.sol [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            uint80 answeredInRound
+        );
+}
+
+
+contract MockAggregatorV2V3 is AggregatorV2V3Interface {
+    uint80 public roundId = 0;
+    uint8 public keyDecimals = 0;
+
+    struct Entry {
```
Comment (justinjmoses) [inline contracts/test-helpers/MockAggregatorV2V3.sol:37]: I'll ask Chainlink to add this to future versions of their interface

**Review by justinjmoses — DISMISSED** [review]
Just a comment about the interface def in the mock.

Additionally, please hold on merging this as #699 can be PRed into here first.

**Review by justinjmoses — CHANGES_REQUESTED** [review]
Blocked pending merge of #699

**Review by justinjmoses — APPROVED** [review]
One remaining comment: https://github.com/Synthetixio/synthetix/pull/715/files#r489461873

**Pull request merged**